test/framework/src/main/java/org/elasticsearch/index/mapper/TestSearchFields.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.apache.lucene.analysis.Analyzer;
+import org.elasticsearch.index.analysis.FieldNameAnalyzer;
+import org.elasticsearch.index.analysis.IndexAnalyzers;
+
+import java.util.Set;
+
+/**
+ * Test implementation of {@link SearchFields} that does not depend on {@link MapperService}.
+ * Each test may override the methods that it needs.
+ */
+public class TestSearchFields extends SearchFields {
+
+    public TestSearchFields() {
```

Comment:
I am open to ideas around testing. I am happy that a lot of mocking is gone, yet I am not entirely happy about I got this done. I was thinking, given that a lot of times only the field type lookup is needed, I could take a function as argument for that common flow.

Reply:
Personally I'd have made `SearchFields` an interface with "production" implementation and the made one or two testing implementations. Like you say, one could take a `Function` for the lookup. I'm not sure this is actually better, but it is what I'd have done.

You could certainly make a static function on this class that takes the `Function` and implements the lookup so that tests that only need the lookup can call that.